### DISCUSSION TITLE
Multiple Stash databases/instances

### DISCUSSION THREAD

**Discussion by UncleRoger33** [stashapp/stash#2703] [Q&A] [answered]
I searched different keywords but couldn't find a similar question in the repository. Sorry if i overlooked and missed one.

Is it possible to use different databases/instances similar to Adobe Lightroom catalogs and switch from one database to another? 

In other words, i'd like to create different databases for different genres. Let's say one database for just professional softcore studios, one for just hardcore, one for just JAV stuff etc. Is there a way or workaround to achieve this?

Thanks.

**Comment by SilentSpartan** [comment]
Yes, it is possible to run multiple instances. I currently run 3. I use Windows though, so I'm not sure how different it is amongst OS's. I just name the .exe's different names and then you need to have a config.yml with updated info on where the new things will be placed. It's been over a year since I set it up so I can't remember completely how you do it, but I'm sure someone on the discord...or here could help you out further.

**Comment by HereIam80** [comment]
I would use three different folders ... one for each instance. The important thing is to just use different ports for each I guess.

**Comment by UncleRoger33** [comment]
Currently using Windows 10. 

I copied **stash-win.exe** to another folder and renamed it to **stash-win2.exe** but couldn't figure a way to duplicate config.yml AKA these parameters without messing the current settings.

![Settings Stash](https://user-images.githubusercontent.com/66418211/176005943-88f21902-a660-4855-85f9-93fa34d3507a.png)

Any ideas?

**Comment by HereIam80** [comment, marked as answer]
Here is what I would do:
Make another folder and put the stash-win.exe and also copy the config.yml in there.
Edit the config.yml with notepad or something and change the following lines:
database:  _New path to your new database_
port: 9998 (or whatever free port you have)

You may change other paths and settings here too if needed. Just don't delete any spaces or change the layout of the file or it can't be read by stash.
Start the exe and browse to http://localhost:9998 or whatever port you chose.
Done.

**Comment by UncleRoger33** [comment]
Thanks for suggestions. I've found a way to do separate collections. Looks a little bit messy but oh well, it works at the end of the day :)


I duplicated the original **config.yml** and modfied the duplicate **config.yml** file based on @HereIam80's post (separate folders for the genre under the main folder and gave a different name to the default **.sqlite** database --> JAV.sqlite)

![solution_01](https://user-images.githubusercontent.com/66418211/176151988-87dffb7d-8789-4923-b419-eeab173b71c6.png)

Then i renamed the original **config.yml** to something else and the modified one to **config.yml**.  Ran stash-win.exe with the modified config.yml and voila! A fresh Stash instance and a JAV.sqlite file was waiting for me.

Fresh Stash instance:
![solution_02a](https://user-images.githubusercontent.com/66418211/176153270-4587b630-ca63-4e09-8f37-9fd8ecf53ef2.PNG)

Fresh database:
![solution_02b](https://user-images.githubusercontent.com/66418211/176153607-b4d79209-e017-4d5b-aad2-14c534fc92b9.PNG)

Settings looks fine too:
![solution_02](https://user-images.githubusercontent.com/66418211/176153800-92f1bacb-109c-4dda-b4e5-c9e8956028a1.png)


From now on i'll just rename and use the relevant **config.yml** files to switch between databases. 

Thanks again for suggestions, appreciated.

**Comment by HereIam80** [comment]
Another hint. You can use different config files with the command `stash-win.exe -c "G:\.stash\cfg\config.yml"`

**Comment by UncleRoger33** [comment]
Oh that's cool. Also i tried your "different stash folders" idea and it worked too 👍  

I am using stash since default config.yml file had to be under `C:\.stash` folder. My database was well populated since then and i hesitated playing with config file. I consider myself a tech-savvy person, but still 😄

**Comment by HereIam80** [comment]
The reason why I would run it in different folders and with different config.yml and .exe files is so I could run multiple instances at once and just browse to the one I need with `http://localhost:9999` or `http://localhost:9998` without the need for switching configs and such.

Anyway ... good you got it working.
If you got more questions don't hesitate to ask. You might get a quicker respone on the discord though as there are many active members around.

**Reply by zelpi** [reply]
I have also the exe-file together with the config.yml in one directory. To create another database, I copied the whole directory and renamed it. Then I changed all the paths in the copied config.yml according to the new directory plus new port number. But when I started the copied exe-file, there‘s an error message, saying that the database can‘t be found. In fact there is the copied database with same name and the new path. When I remove it, same error. I suppose, the system should create somehow a new database. Any help, please…?

**Reply by UncleRoger33** [reply]
Can you upload a screenshot of your config.yml file?

**Reply by zelpi** [reply]
Thanks for the quick reaction. A second db would have been for my music videos. Still not resolved but I decided therefore to use another software. However stash remains a really good program for it‘s original purpose. The option for a second db might be implemented in the future via menu within the program…

**Comment by I3Jb0i** [comment]
I’m running two instances in docker and it works great.  I set the same login credentials on both, so I only have to log in once and use links to hop between them

**Comment by miaLDN** [comment]
Do the multiple servers share the same database?
